lib/Runtime/PlatformAgnostic/Platform/Linux/NumbersUtility.cpp

```diff
@@ -0,0 +1,267 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Common.h"
+#include "ChakraPlatform.h"
+#include <locale.h>
+
+namespace PlatformAgnostic
+{
+namespace Numbers
+{
+    #define BufSize 256
+
+    template<class T, class N=T>
+    void ByteCopy(const T *from, N *to, const size_t max_length)
+    {
+        size_t length = 0;
+        for(;length < max_length && (to[length] = static_cast<N>(from[length])) != 0;
+            length++);
+        if (length == max_length)
+        {
+            to[length-1] = N(0);
+        }
+    }
+
+    static const WCHAR number_zero        = WCHAR('0');
+    static const WCHAR number_five        = WCHAR('5');
+    static const WCHAR number_nine        = WCHAR('9');
+
+    Utility::NumbersLocale::NumbersLocale()
+    {
+        maxDigitsAfterDecimals = 3;
+        defaultDecimalDot = WCHAR('.');
+        defaultDecimalComma = WCHAR(',');
+        localeNegativeSign = WCHAR('-');
+        
+        char buffer[BufSize];
+        char *old_locale = setlocale(LC_NUMERIC, NULL);
+
+        if(old_locale != NULL)
+        {
+            ByteCopy<char>(old_locale, buffer, BufSize);
```

Comment:
Should be fine but would be nice to add an assert here that strlen(old_locale) < BufSize